src/examples/boost_multi_array_2D.cpp

```diff
@@ -18,8 +18,8 @@ using namespace HighFive;
 
 const std::string FILE_NAME("boost_multiarray_example.h5");
 const std::string DATASET_NAME("dset");
-const size_t size_x = 10;
-const size_t size_y = 3;
+const long size_x = 10;
```

Comment:
Feeling puzzled. It doesnt warn without the L like `10l`
Anyway why not use the good-old & fast `int` type for small loops?

Reply:
It doesn't warn because an integer fits in a long, and the compiler makes the type conversion implicitly.

Reply:
Besides, I changed to `int` as suggested